src/components/LocalNav/LocalNav.hooks.tsx

```diff
@@ -1,2524 +1,68 @@
-        id: 15,
-        position: 15,
-        learning_element_id: 15,
-        learning_path_id: 15,
-        recommended: false,
-        learning_element: {
-          id: 15,
-          lms_id: 15,
-          name: 'Retrospektive Reihenfolge der Lernelemente',
-          activity_type: 'feedback',
-          classification: 'RQ',
-          university: 'HS-KE',
-          created_by: 'me',
-          created_at: 'now',
-          last_updated: 'now',
-          student_learning_element: {
-            id: 15,
-            student_id: 15,
-            learning_element_id: 15,
-            done: false,
-            done_at: 'now'
-          }
-        }
-      },
-      {
-        id: 16,
-        position: 16,
-        learning_element_id: 16,
-        learning_path_id: 16,
-        recommended: false,
-        learning_element: {
-          id: 16,
-          lms_id: 16,
-          name: 'Abschließende Fragen',
-          activity_type: 'feedback',
-          classification: 'RQ',
-          university: 'HS-KE',
-          created_by: 'me',
-          created_at: 'now',
-          last_updated: 'now',
-          student_learning_element: {
-            id: 16,
-            student_id: 16,
-            learning_element_id: 16,
-            done: false,
-            done_at: 'now'
-          }
-        }
-      }
-    ]
-  }
-]
+    }
+  ]
+}
 
 export const getSortedLearningPath = async (
-  data: Topic[],
-  userId: number,
-  lmsUserId: number,
-  studentId: number,
-  fetchLearningPath: LearningPathReturn
-): Promise<LearningPath[]> => {
-  const promises = data.map((topic) => fetchLearningPath(userId, lmsUserId, studentId, "2", String(topic.id))) //reihenfolge der parameter beachten
-  const learningPaths = await Promise.all(promises)
-
-  return learningPaths.map((learningPath) => {
-    learningPath.path.sort((a, b) => a.position - b.position)
-    return learningPath
-  })
+  userid: number,
+  lmsUserid: number,
+  studentid: number,
+  data: Topic,
+  fetchLearningPath: LearningPathElementReturn
+): Promise<LearningPathElement> => {
+  const learningPath = await fetchLearningPath(userid, lmsUserid, studentid, '2', data.id.toString())
+  learningPath.path.sort((a, b) => a.position - b.position)
+  return learningPath
 }
 
-export const useLearningPath = (): { loading: boolean; topics: Topic[]; learningPaths: LearningPath[] } => {
+export const useLearningPathTopic = (): { loading: boolean; topics: Topic[] } => {
   const [loading, setLoading] = useState(true)
   const [topics, setTopics] = useState<Topic[]>([])
-  const [learningPaths, setLearningPaths] = useState<LearningPath[]>([])
-  //const fetchUser = useBoundStore((state) => state.fetchUser)
-  //const fetchLearningPath = useBoundStore((state) => state.fetchLearningPath)
+  const fetchUser = usePersistedStore((state) => state.fetchUser)
+  const fetchLearningPathTopic = useStore((state) => state.fetchLearningPathTopic)
 
   const effect = async () => {
     setLoading(true)
```

Comment:
Man kann das effect() auch in der useEffect zu definieren, dann ist es nur in diesem Context.
```suggestion
    const effect = async () => {
```